include/aws/crt/StlAllocator.h

```diff
@@ -47,7 +48,11 @@ namespace Aws
             {
                 (void)hint;
                 AWS_ASSERT(m_allocator);
-                return reinterpret_cast<RawPointer>(aws_mem_acquire(m_allocator, n * sizeof(T)));
+                if (reinterpret_is_safe())
+                {
+                    return reinterpret_cast<RawPointer>(aws_mem_acquire(m_allocator, n * sizeof(T)));
+                }
+                return static_cast<RawPointer>(aws_mem_acquire(m_allocator, n * sizeof(T)));
```

Comment:
would it be safe to just ALWAYS do static_cast?
wouldn't that automagically compile to be equivalent to reinterpret_cast, if it's safe to do reinterpret_cast?

Reply:
Static cast seems to be the way to go!